src/main/scala/com/cloudera/sparkts/models/ARIMAX.scala

```diff
@@ -476,7 +476,15 @@ class ARIMAXModel(
     var error = 0.0
 
     val xregMatrix = new BreezeDenseMatrix(rows = xreg.flatten.length / xreg.size, cols = xreg.size, data = xreg.flatten)
-    val xregPredictors = AutoregressionX.assemblePredictors(ts.toArray, MatrixUtil.matToRowArrs(xregMatrix), 0, xregMaxLag, includeOriginalXreg)
```

Comment:
There are 4 tests from 32 which fails (25,26,27,28). They fail during assembling predictors - the length of time series is smaller then the length of xreg because ts is after differencing and lagging by "d" value and xreg is just 1. differentiated 2. convert to matrix and pass into assemblePredictors method.